### ISSUE TITLE
No coffee???

### ISSUE THREAD

**Issue by fribse** [nielsfaber/alarmo#88]
Hi Niels (wow that sounds terribly danish?)

I have a big issue with your github, there is no 'buy me a coffee' or the like button???
This alarm card add on is incredible, it's way above and beyond.
Sleak, easy to understand and a lot of features.
The key feature being multiple pin codes, I simply can't grasp that HA only has one, and rely on the alarm arming and disarming by itself.
**Features** is maybe in need of trimming in my opinion, I know it's tempting to just add features, but please remember the eco system and utilize that.
Some of the built in automations would fit just as well in the regular automations, and would not limit people to migrate away from your alarmo to something else in the future if you stop your project.

I have a node-red flow for my old alarm, which would never fit in the alarmo automations, and now I'm in a situation where something has to be moved to alarmo (failed to arm), and others would be better off in node-red, and I will adopt it to react on the states of the alarmo. This way I'm not tied down to your system completely.

Oh, and an enhancement request, if you want to keep the 'actions' tab, which is perfect, it should allow to turn on and off of groups in the UI. I have several Heiman Zigbee Sirens, and they are in a group, and it's simply silly to create multiple actions for the sirens, so I've modified the YAML to do homeassistant.turn_on /off for the group. It would be nice to have that directly in the UI.

**Comment by nielsfaber** [comment]
Hello Kenneth, 
Thanks for appreciating Alarmo that much and offering a donation.
If you insist, here's my [buymeacoffee](https://www.buymeacoffee.com/vrdx7mi) page. Thank you in advance.

I am not that satisfied with Alarmo yet, there are quite some things I want to tackle.
Its definitely not done 😄 
The documentation is one of the things that deserves more attention.

The automations part is a bit of a struggle. I'm overwhelmed by the amount of feature requests for it, which would make it a duplicate of the HA automations engine, which was not the intention. 
Maybe it should be replaced by a set of blueprints that allow users to set up most used stuff.

Could you elaborate on this:

> Features is maybe in need of trimming in my opinion, I know it's tempting to just add features, but please remember the eco system and utilize that.

Feedback is much appreciated since my own house alarm is quite simple. Practical use-cases help the further development.
Could you explain a bit about your setup? Why do you have part in Alarmo and part in node-red?
Could things be improved to make it more maintainable?

Thanks.

**Comment by fribse** [comment]
Oh, I do insist 😆 
The link should be in the readme.md 😉 

Well, with trimming, I was referring to the actions part, I think a set of Blueprints would be fantastic utilization of the ecosystem.

I can show you the node-red setup that I've currently deactivated while contemplating what I want to do now, please note that I'm not a programmer, so others will probably make it smarter 😄 
![image](https://user-images.githubusercontent.com/13660621/104088648-b748a580-5268-11eb-8aea-05b02bff52e2.png)
The flow uses much of the same logic for all actions, just replaces the 'message' that is sent.

My setup isn't that complicated. I'm replacing a professional alarm setup with the HA alarm, as it's silly to have two sensors on doors and windows. My primary concern with the HA standard alarm is that there is only one pin code. We have housesitters when we travel, we have cleaners sometimes, and they need their own pin, we also have children, both living at home and away, they also need codes.

Regarding the hw setup.
I have a lot of windows / doors sensors.
All zigbee based, and tied into HA with deconz.

I normally have several modes (most of them you have already), home/night/out/off/panic (last one is so far missing).
The Panic flow I've created is a bit tricky, it will announce in the speakers that it is deactivated, but then wait 10s and send a panic message to the cellphones, panic is meant for intruders, and should therefore obfuscate that it is alerting people.
I was also thinking that a 'please help' is normal to have in an alarm (call for 'hjemmehjælp' or the likes), it's not that big a deal to implement that as a seperate service though.

One thing I always do, is announce in all the speakers (group) that the alarm status has changed (as well as the two panels I have), and I also send to cell's about that (last thing is 'under revision' :grin ).
As I'm using Android, I have added specific 'buzz' patterns and icons colours to force messages even if the phone is off there are also added a few extra settings to the message.

```
{
   "data": {
     "message": "Alarmen er {{ payload }}",
     "title": "Alarm er {{ payload }}",
     "data": {
        "tag": "alarm",
        "color": "{{ color }}",
        "channel": "alarm",
        "importance": "high",
        "vibrationPattern": "1000, 1000, 1000, 1000, 1000, 1000",
        "ttl": 0,
        "priority": "high"
     }
   }
}
```

I also have two panels in the house (tablets), one is at the entry and is only showing the alarm panel. The other one is in the livingroom, and my idea was that if the alarm is activated, the window it shows should change to a specific alarm panel lovelace view.

I'm going to make a flow also that plays a 'ping' every 15 s or something like that, in the panel at the entry after all the announcements, during exit and entry in the delayed zone (whatever it's called).

We also have smoke detectors, which would also fit nicely in an alarm setup (our old pro alarm did that). But the flow for that is way more complicated.

![image](https://user-images.githubusercontent.com/13660621/104089506-fd553780-526f-11eb-91ce-269a315cd986.png)

The complication comes from the challenge, that I want the speakers to announce where they are triggered, and if there are more than one, it should say 'and' between them, so it could be 'there is smoke in Adrians room and Stairs'.
It also flashes the lights in red in the childrens rooms, and sends alerts to cellphones. After 1 minute it starts the 'big' sirens, and turns up all the light to 100% (unless the alarm is deactivated).

**Comment by nielsfaber** [comment]
Thank you very much for taking the time to write down how your setup looks like.

I was planning to add a panic mode at some point (when the foundation is stable and documentation is improved).
Currently you can already configure some binary_sensor as 'always-on' type, such that it would always result into triggering.
What's missing is the ability to program a dedicated response, either by having a separate 'panic' state or by introducing some attribute, and trigger automations on this event.
Additionally, I could have Alarmo generate a panic switch entity, or introduce a service for this.
I would like to gather opinions/use-cases by the time I find the time to implement this.

The notifications part in Alarmo is something I would like to improve.
It seems that a lot of users use speech/text notifications.
HA has some internal translations for different sensor types+states, like "window is open", I'm investigating if these can be re-used to help formatting a meaningful message.
Also I'm aware that the Actions UI currently does not give enough functionality for supporting all notification integrations, I will either extend it or find an alternative.

The flashing lights + delayed siren activation + ping sound every X seconds, many users are using or looking for such functions.
It would be great if Alarmo could simplify setting such things up, since it seems everyone is re-inventing the wheel here.
I'm leaning towards having some 'library' of blueprints listed within the Alarmo panel (I see the blueprints section in the HA community forums already becoming a mess).
Ideally these blueprints can be installed from here as well, but I'm not sure if something like this is possible.
I will probably keep the Action UI as it is for now, and replace it by something entirely different (as it only really offers to switch on/off a device currently).

> Btw. I would love to use the AlarmPanel by Thanksmister on the entry way tablet, but it seems to only handle one pin :-(

As far as I know, this panel uses MQTT to connect to HA.
Currently quite some things are hard-coded (like delay settings and the available arm modes). 
I have reviewed a PR of Thanksmister in HA that would eliminate these restrictions, so things should get better over time.
My expectation would be that the PIN is handled by HA entirely, so if Alarmo would be connected it should work with multiple pins.
I'm not sure where the restriction is exactly... But I agree that this panel is a good companion for Alarmo.
There are also some ideas for creating a custom card that extends the standard Lovelace Alarm Panel card with some extra functions.
Like options for arming forcibly, skipping delay, and maybe some provisions for handling multiple area setups.

Anyway, thanks for giving Alarmo a try, I hope to improve things for you.
I propose to stop this discussion here, and continue in a larger setting.
I would like to use the forums for collecting opinions before starting a new feature.
You're also invited to open feature requests here, or share your views on the existing ones.
